mantidproject/mantid
# Fix error reporter launcher under Python 3

**Description of work.**

Fixes the executable used to start the error reporter dialog. It pulls the executable value from the cmake variable set in the configuration. Hard coding the executable to Python 2 breaks starting the dialog under Python 3. An example of this is a [recent build failure](http://builds.mantidproject.org/job/pull_requests-ubuntu-python3/9869/console):

```
reading sources... [  3%] algorithms/CalculateIqt-v1
ApplicationWindow-[Fatal] Fatal error on line 223 of "runsphinx_html.py" encountered:
ApplicationWindow-[Fatal] SystemExit: 1
ApplicationWindow-[Fatal]   at line 223 in 'runsphinx_html.py'
ApplicationWindow-[Fatal]   caused by line 108 in 'runsphinx_html.py'
MantidUI-[Notice] MantidPlot is shutting down...

Exception occurred:
  File "/home/builder/Jenkins/workspace/pull_requests-ubuntu-python3/docs/sphinxext/mantiddoc/directives/diagram.py", line 88, in execute
    raise RuntimeError("Cannot find dot-file: '" + diagram_name + "' in '" + os.path.join(env.srcdir,"diagrams"))
RuntimeError: Cannot find dot-file: 'TransformToIqt-v1_wkflw.dot' in '/home/builder/Jenkins/workspace/pull_requests-ubuntu-python3/docs/source/diagrams
The full traceback has been saved in /tmp/sphinx-err-weao8b72.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
Traceback (most recent call last):
  File "/home/builder/Jenkins/workspace/pull_requests-ubuntu-python3/scripts/ErrorReporter/error_dialog_app.py", line 15, in <module>
    import mantid # noqa
  File "/home/builder/Jenkins/workspace/pull_requests-ubuntu-python3/build/bin/mantid/__init__.py", line 25, in <module>
    from . import pyversion
  File "/home/builder/Jenkins/workspace/pull_requests-ubuntu-python3/build/bin/mantid/pyversion.py", line 18, in <module>
    raise ImportError(message % (TARGET_VERSION, _running_vers))
ImportError: Python version mismatch, cannot continue.
Mantid was built against version '3.5' but you are running version '2.7'. These versions
must match.
```

**To test:**

* Launch MantidPlot
* run the Segfault algorithm
* the error dialog should appear (as long as `usagereports.enabled = 1` is set in the config)

This should ideally be tested under both Python 2 and Python 3.

*No associated issue*

Does this update require release notes?


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
